### ISSUE TITLE
Is possible that number of records in chunk exceeds buffer_chunk_records_limit?

### ISSUE THREAD

**Issue by mia-0032** [tagomoris/fluent-plugin-buffer-lightening#6]
Hi, I use fluent-plugin-buffer-lightening in https://github.com/mdoi/fluent-plugin-gcloud-pubsub .

Default buffer_chunk_records_limit in fluent-plugin-gcloud-pubsub is `900`.
But I encountered a error that the chunk has over 900 records.

Two fluentd are runnning my environment: (1) tail logs and forward another fluentd, (2) send logs to Google Cloud PubSub.

This error seem to occur when buffer chunk is large on fluentd(1).
Is possible that number of records in chunk exceeds buffer_chunk_records_limit?

Error messages is below:

```
[error]: plugin/out_gcloud_pubsub.rb:60:rescue in write: unexpected error error="3:The value for message_count is too large. You passed 3180 in the request, but the maximum value is 1000."
```

(1) fluent.conf is below:

```
<source>
  @type tail
  path /path/to/dummy.log
  pos_file /path/to/dummy.log.pos
  tag test.local
  format ltsv
</source>

<match test.**>
  @type forward
  <server>
    name local-fluentd
    host 127.0.0.1
    port 24225
  </server>
</match>
```

(2) fluentd.conf is below:

```
<source>
  @type  forward
  port   24225
</source>

<match test.**>
  @type gcloud_pubsub
  project my-project
  topic my-topic
  key /path/to/key.json
</match>
```

**Comment by wuub** [comment]
@mia-0032 :+1: it seems to happen when forwarding agent (1) sends data in large batches (as it almost always does in production). In this case your central (2) agent can get lots of data in one go and easily fill buffer from 0 to much much more than allowed 900 records. 

https://github.com/tagomoris/fluent-plugin-buffer-lightening/blob/master/lib/fluent/plugin/buf_lightening.rb does not seem handle that  situation as expected.

**Comment by tagomoris** [comment]
Yes. This buffer plugin cannot handle such situation perfectly, and it's brought by Fluentd's API limitation.
This limitation is solved at Fluentd v0.14, and I'll introduce more improvements to handle such cases into built-in buffer plugins.

**Comment by mia-0032** [comment]
@wuub @tagomoris  I see that the error occurs due to Fluentd's API limitation. I'll limit record flow in the plugin and improve for upgrading to v0.14.0. Thank you for your replies!
